### DISCUSSION TITLE
Debian Packaging discussions

### DISCUSSION THREAD

**Discussion by hwspeedy** [hardinfo2/hardinfo2#16] [General]
DEBIAN PACKAGING DISCUSSION

**Comment by hwspeedy** [comment]
@tsimonq2 - Do you want to package hardinfo2 for Debian?

**Reply by hwspeedy** [reply]
@lucascastro     CC: @lpereira
Regarding packaging of hardinfo2 into Debian - I have had no luck contacting @tsimonq2. - It probably means he is too busy for a community edition of hardinfo.

Above is the organization starting to build up: https://github.com/hardinfo2/hardinfo2/discussions/16

We talked about the package name on oftc debian mentors:
 - Fedora has been submitted with package name hardinfo2 by @topazus - which was also what some of the other debian mentors said to do for Debian. 
The branding of community edition is hardinfo2 (due to what was possible) and both Fedora and Debian support upgrading other package names. 

Please note that we have made it possible to make packages for up to ~10 years old distros and would like to offer them an update as well.

When you have the time for Debian packaging - please create a new issue with debian packaging, thanx.

Here is fedora packaging coordination thread: https://github.com/hardinfo2/hardinfo2/issues/15
Done

Here is release page with status for debian/fedora: https://github.com/hardinfo2/hardinfo2/releases/tag/release-2.0.15

/hwspeedy

**Comment by lucascastro** [comment]
Em 09/03/2024 19:41, hwspeedy escreveu:
>
> @lucascastro <https://github.com/lucascastro> CC: @lpereira 
> <https://github.com/lpereira>
> Regarding packaging of hardinfo2 into Debian - I have had no luck 
> contacting @tsimonq2 <https://github.com/tsimonq2>. - It probably 
> means he is too busy for a community edition of hardinfo.
>
That's not good.
>
> Above is the organization starting to build up: #16 
> <https://github.com/hardinfo2/hardinfo2/discussions/16>
>
> We talked about the package name on oftc debian mentors:
>
>   * Fedora has been submitted with package name hardinfo2 by @topazus
>     <https://github.com/topazus> - which was also what some of the
>     other debian mentors said to do for Debian.
>     I do not know what is possible or how. The branding of community
>     edition is hardinfo2 (due to what was possible) and both Fedora
>     and Debian support upgrading other package names.
>     It's entirely up to fedora and debian to agree on package name and
>     model.
>
I don't guess hardinfo2 a good name, maybe hardinfo-ng (hardinfo next 
generation).

But hardinfo2 is forked from (old) hardinfo, what I really think about 
this issue it'd be good if lpereire point the hardinfo2 repository as a 
new official upstream repository, so it'd be good for hardinfo package 
update from the new one repository.

> @mckaygerhard <https://github.com/mckaygerhard> fears that some one 
> else would continue hardinfo - don't really understand. (See last of 
> fedora packaging thread)
>
> Please note that we have made it possible to make packages for up to 
> ~10 years old distros and would like to offer them an update as well.
>
That's because there's no information about the new official upstream 
repository.
>
> When you have the time for Debian packaging - please create a new 
> issue with debian packaging, thanx.
>
No thing to do about the old package until the official upstream 
repository be set to hardinfo2.

What it'd be done is the hardinfo2 package setting solve 
conflicts/problem on the old/other one.

> Here is fedora packaging coordination thread: #15 
> <https://github.com/hardinfo2/hardinfo2/issues/15>
> 4 patches needed so far. Describes in thread and release page.
>
> Here is release page with status for debian/fedora: 
> https://github.com/hardinfo2/hardinfo2/releases/tag/release-2.0.12
>
> /hwspeedy
>
> —
> Reply to this email directly, view it on GitHub 
> <https://github.com/hardinfo2/hardinfo2/discussions/16#discussioncomment-8732332>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AALJN6WPNRHZGCDX5DH737TYXOFX7AVCNFSM6AAAAABEK7KGHGVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DOMZSGMZTE>.
> You are receiving this because you were mentioned.
>

Thanks,

Lucas Castro.

**Reply by lpereira** [reply]
I'm probably going to rename the `master` branch in lpereira/hardinfo to `main-pre-community-edition` and push the `main` branch of hardinfo2/hardinfo2 to be the `main` branch in my personal repo.

I've been pretty busy with life and work and haven't had a whole lot of time and enthusiasm to work on any of my open source projects, but I'm happy that there are people out there that want to continue hardinfo and finally push a new stable version out after over a decade.  (I'll be around the project but it's unlikely I'll actively participate as other things have taken priority in my life.)

I'm not married to any name; in fact, we could potentially rename this organization to `hardinfo` instead and it might make more sense and help move the ball for all the downstream packagers.  There's no reason to call it `hardinfo2` or `hardinfo-ng` or anything of the sort, as this is pretty much a continuation of the old project.

(A bit unrelated to the discussion: as far as the domain `hardinfo.org` goes, however, I think that the page could be hosted here on GitHub, using GitHub pages, as it's easy to publish, free, well-maintained, and well-integrated with the actions feature.  The current `api.hardinfo.org` server is written in Go and runs on a VPS I control, and I'd like to keep using that as it provides us good control of what is cached and when it's refreshed, all the while being pretty stable and already running for a good while.  I already have monitoring and automated offsite backup set up for that too, which is good.  I can also offer `@hardinfo.org` email aliases to developers if they express the desire to.)

**Comment by lucascastro** [comment]
Em 13/03/2024 17:20, L. Pereira escreveu:
>
> I'm probably going to rename the |master| branch in lpereira/hardinfo 
> to |main-pre-community-edition| and push the |main| branch of 
> hardinfo2/hardinfo2 to be the |main| branch in my personal repo.
>
If possible, make a release on github or somewhere else, so it'll be 
better to prepare package distribution upgrade.
>
> I've been pretty busy with life and work and haven't had a whole lot 
> of time and enthusiasm to work on any of my open source projects, but 
> I'm happy that there are people out there that want to continue 
> hardinfo and finally push a new stable version out after over a 
> decade. (I'll be around the project but it's unlikely I'll actively 
> participate as other things have taken priority in my life.)
>
You had been working on this by year, by now if you just merge changes 
and prepare release tarballs or set tags for releases, it'll be good 
enough.
>
> I'm not married to any name; in fact, we could potentially rename this 
> organization to |hardinfo| instead and it might make more sense and 
> help move the ball for all the downstream packagers. There's no reason 
> to call it |hardinfo2| or |hardinfo-ng| or anything of the sort, as 
> this is pretty much a continuation of the old project.
>
Agreed on this.
>
> (A bit unrelated to the discussion: as far as the domain 
> |hardinfo.org| goes, however, I think that the page could be hosted 
> here on GitHub, using GitHub pages, as it's easy to publish, free, 
> well-maintained, and well-integrated with the actions feature. The 
> current |api.hardinfo.org| server is written in Go and runs on a VPS I 
> control, and I'd like to keep using that as it provides us good 
> control of what is cached and when it's refreshed, all the while being 
> pretty stable and already running for a good while. I already have 
> monitoring and automated offsite backup set up for that too, which is 
> good. I can also offer ***@***.***| email aliases to developers if 
> they express the desire to.)
>
That domain it'd be a good places for news updates and releases hosting, 
so someone can take care of releases tarballs, gpg release signature 
therefore you or someone else just publish over there.


Let me know when the merge is done.

> —
> Reply to this email directly, view it on GitHub 
> <https://github.com/hardinfo2/hardinfo2/discussions/16#discussioncomment-8778316>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AALJN6XFV46YJFV3KNSBZ2DYYCYIJAVCNFSM6AAAAABEK7KGHGVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DONZYGMYTM>.
> You are receiving this because you were mentioned.
>

Lucas Castro.

**Comment by hwspeedy** [comment]
> I'm probably going to rename the `master` branch in lpereira/hardinfo to `main-pre-community-edition` and push the `main` branch of hardinfo2/hardinfo2 to be the `main` branch in my personal repo.
> 
> I've been pretty busy with life and work and haven't had a whole lot of time and enthusiasm to work on any of my open source projects, but I'm happy that there are people out there that want to continue hardinfo and finally push a new stable version out after over a decade. (I'll be around the project but it's unlikely I'll actively participate as other things have taken priority in my life.)

**So why should the community project be behind your personal repo for debian?  
That was one of the issues at the beginning, that it was difficult to really do all the updates needed when you had very little time.
This program can out live us all and hardinfo2 is what was possible - So please consider setting hardinfo2 free as community edition and change upstream for debian.**
 
> I'm not married to any name; in fact, we could potentially rename this organization to `hardinfo` instead and it might make more sense and help move the ball for all the downstream packagers. There's no reason to call it `hardinfo2` or `hardinfo-ng` or anything of the sort, as this is pretty much a continuation of the old project.

**hardinfo2 is a reboot but also a next edition and is already not 100% compatible with hardinfo.  But I see no problem releasing it as hardinfo with alias hardinfo2 in debian or what ever you like.**

> (A bit unrelated to the discussion: as far as the domain `hardinfo.org` goes, however, I think that the page could be hosted here on GitHub, using GitHub pages, as it's easy to publish, free, well-maintained, and well-integrated with the actions feature. The current `api.hardinfo.org` server is written in Go and runs on a VPS I control, and I'd like to keep using that as it provides us good control of what is cached and when it's refreshed, all the while being pretty stable and already running for a good while. I already have monitoring and automated offsite backup set up for that too, which is good. I can also offer `@hardinfo.org` email aliases to developers if they express the desire to.)

**I see a great potential in web development and webapp development to get more FOSS developers interested. This project is basic and allows anyone to learn and develop skills.
We are going to make this hardinfo2 webapp through github server repo.** 

**The VPS in this project is easy to deploy and gives features and functions not found on a simple github.io. 
The backup is github - No offsite backup not available to all nor missing database, setup, etc. like the old repo.
This is all in github. So that is what I will continue for hardinfo2.**

**Please set hardinfo2 free, thanx.**

**Comment by hwspeedy** [comment]
@lpereira @lucascastro - RESEND WITH MENTIONING.

> I'm probably going to rename the `master` branch in lpereira/hardinfo to `main-pre-community-edition` and push the `main` branch of hardinfo2/hardinfo2 to be the `main` branch in my personal repo.
> 
> I've been pretty busy with life and work and haven't had a whole lot of time and enthusiasm to work on any of my open source projects, but I'm happy that there are people out there that want to continue hardinfo and finally push a new stable version out after over a decade. (I'll be around the project but it's unlikely I'll actively participate as other things have taken priority in my life.)

**So why should the community project be behind your personal repo for debian?  
That was one of the issues at the beginning, that it was difficult to really do all the updates needed when you had very little time.
This program can out live us all and hardinfo2 is what was possible - So please consider setting hardinfo2 free as community edition and change upstream for debian.**
 
> I'm not married to any name; in fact, we could potentially rename this organization to `hardinfo` instead and it might make more sense and help move the ball for all the downstream packagers. There's no reason to call it `hardinfo2` or `hardinfo-ng` or anything of the sort, as this is pretty much a continuation of the old project.

**hardinfo2 is a reboot but also a next edition and is already not 100% compatible with hardinfo.  But I see no problem releasing it as hardinfo with alias hardinfo2 in debian or what ever you like.**

> (A bit unrelated to the discussion: as far as the domain `hardinfo.org` goes, however, I think that the page could be hosted here on GitHub, using GitHub pages, as it's easy to publish, free, well-maintained, and well-integrated with the actions feature. The current `api.hardinfo.org` server is written in Go and runs on a VPS I control, and I'd like to keep using that as it provides us good control of what is cached and when it's refreshed, all the while being pretty stable and already running for a good while. I already have monitoring and automated offsite backup set up for that too, which is good. I can also offer `@hardinfo.org` email aliases to developers if they express the desire to.)

**I see a great potential in web development and webapp development to get more FOSS developers interested. This project is basic and allows anyone to learn and develop skills.
We are going to make this hardinfo2 webapp through github server repo.** 

**The VPS in this project is easy to deploy and gives features and functions not found on a simple github.io. 
The backup is github - No offsite backup not available to all nor missing database, setup, etc. like the old repo.
This is all in github. So that is what I will continue for hardinfo2.**

**Please set hardinfo2 free, thanx.**

**Reply by lpereira** [reply]
Oh, no, that's not what I said!  What I meant was: my personal repo will exist solely as a soft-fork  (i.e. a mirror, for personal archiving reasons) of the main development repository as part of the hardinfo organization. I'm going to move the current `master` branch in my personal repo to another name and push (and try to keep up-to-date) the `main` branch from the organization repo to my personal repo, but that's it. I'll even disable filing of issues and PRs in my personal repo, with a message to go over to the organization one.

**Comment by hwspeedy** [comment]
@lpereira @lucascastro 

Okay, That sounds a lot more like, what we talked about, thanx for clarifying.

**Debian upstream is set to hardinfo2.**
branding of community project is hardinfo2
@lucascastro finds the package settings to support users the best and allow to install as hardinfo2 also.

**Private repo**
I would love is you just made a single commit to README.md with a point to hardinfo2 and then made it read/only archived.
We will be making history pages on the hardinfo2 to preserve history from when linux was young.

**Private hardinfo domain**
if possible this would help the project most:
- www.hardinfo.org CNAME www.hardinfo2.org
- hardinfo.org CNAME hardinfo2.org
- api.hardinfo.org CNAME api.hardinfo2.org

Can we all agree on this?

**Comment by lucascastro** [comment]
Em 14/03/2024 14:53, hwspeedy escreveu:
>
> @lpereira <https://github.com/lpereira> @lucascastro 
> <https://github.com/lucascastro>
>
> Okay, That sounds a lot more like, what we talked about, thanx for 
> clarifying.
>
> *Debian upstream is set to hardinfo2.*
> branding of community project is hardinfo2
> @lucascastro <https://github.com/lucascastro> finds the package 
> settings to support users the best and allow to install as hardinfo2 also.
>
Right now debian package monitor upstream releases on lpereira repository.
>
> *Private repo*
> I would love is you just made a single commit to README.md with a 
> point to hardinfo2 and then made it read/only archived.
> We will be making history pages on the hardinfo2 to preserve history 
> from when linux was young.
>
Therefore if lpereira point somewhere, in README should be good, the 
upstream repository had moved  to hardinfo2 it'd be good.


The merge lpereira just talked about it. It's not that bad and it 
doesn't turn the community reboot impossible.

I look at the merge as smoothly transition. So lpereira could merge and 
tag a release or just point the new upstream repository.


For GNU/Linux distribution point to a new repository will improve the 
update from old one package, who'll take care for packaging, maintenance 
and update

could package and release update with no need to make it from start and 
get through all the way process of new package introduction.


> *Private hardinfo domain*
> if possible this would help the project most:
>
>   * www.hardinfo.org <http://www.hardinfo.org> CNAME www.hardinfo2.org
>     <http://www.hardinfo2.org>
>   * hardinfo.org CNAME hardinfo2.org
>   * api.hardinfo.org CNAME api.hardinfo2.org
>
> Can we all agree on this?
>
If the project get some place  for announcement, releases news, etc... 
it's well done.
>
> —
> Reply to this email directly, view it on GitHub 
> <https://github.com/hardinfo2/hardinfo2/discussions/16#discussioncomment-8789767>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AALJN6RBB5ZHRHWR6X3UAKLYYHPYPAVCNFSM6AAAAABEK7KGHGVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DOOBZG43DO>.
> You are receiving this because you were mentioned.
>

Lucas Castro.

**Comment by hwspeedy** [comment]
@lucascastro @lpereira 

>If the project get some place  for announcement, releases news, etc...
>it's well done.

**webpage:**
https://hardinfo2.org is the user entry page.
comes from https://github.com/hardinfo2/server/tree/main/www

Please send example what you would like on it and I will make it happen.

**Release with text:**
https://github.com/hardinfo2/hardinfo2/releases/tag/release-2.0.15

Please start a new issue for debian packaging on hardinfo2, thanx.

**Maintainer**
I will be doing the maintainer/release/packaging help to you/bug fixing/updating/etc.
 - Just like with fedora/redhat - which is in distros: https://github.com/hardinfo2/hardinfo2/issues/15

PS: lpereira has the same rights to everything on hardinfo2 - she is backup owner on community project to ensure survival if anything happens.

**Comment by lpereira** [comment]
Just updated my personal ``hardinfo`` repo to have a note in the ``README.md`` file pointing to this repo as the current development repository for this project.  Hopefully that serves as the note that @lucascastro mentioned. :)

**Comment by hwspeedy** [comment]
@lucascastro @tsimonq2

We have added a new version of the original GUI drawing benchmark for fun and historical values -> down to one open wish in the Debian bug tracker system.

https://github.com/hardinfo2/hardinfo2/releases/tag/release-2.0.18

Is Ubuntu releases through the Debian building systems? Will you be able to do that also?

If there is anything I can do - just let me know, thanx.

In advance thanx.

**Comment by hwspeedy** [comment]
@lucascastro @tsimonq2 

According to Debian tracker - **the old hardinfo package will be removed from Debian very soon 2024.04.11**, as it did not have GTK3 and LibSoup3 along with all the unfixed errors.

https://tracker.debian.org/pkg/hardinfo

@lucascastro you said that you wanted to continue on the hardinfo package instead of a new submission and thereby clean the bug tracker system, which is possible except for one wish, that should remain open (https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=487835).

The rebooted hardinfo2 has all the bug fixes, etc. and now has GTK3 and LibSoup3 as standard choice for distros that support this.

@lpereira has made this hardinfo2 repo, the upstream for hardinfo (https://github.com/lpereira/hardinfo/blob/master/README.md), so everything should be ready for releasing hardinfo2 in Debian & Ubuntu.

**Comment by hwspeedy** [comment]
**Debian/Ubuntu Release to official repos - UPDATE**

Communication is done by email - preferred by @lucascastro 

@tsimonq2 has not replied to neither lucascastro nor me.

@lucascastro is doing an ITP for hardinfo2, due to no answer.

CC:@lpereira 

PS: New release 2.1.2 with fix from @lucascastro - https://github.com/hardinfo2/hardinfo2/releases/tag/release-2.1.2

**Comment by hosiet** [comment]
So here is something new:

* I intend to take over the package maintenance of hardinfo ( https://tracker.debian.org/pkg/hardinfo ) and replace @tsimonq2 via Debian's internal process. This is being tracked in https://bugs.debian.org/1071373 . Due to timeout requirement set by Debian's policy, the maintenance takeover will happen after 1 month if @tsimonq2 does not response.
* In the meanwhile, I don't mind spending some time to play around hardinfo2's codebase and find any potential issues early.
* I am a Debian Developer, aka official Debian member.
* I could not find an ITP for hardinfo2 yet in Debian's database. cc @lucascastro 
* After that, I would like to cooperate with hardinfo2 upstream and decide the best package name. Currently I think both "hardinfo" or "hardinfo2" would be okay, depending on which name is the most popular among Linux users. If you want, I can make package "hardinfo" to provide "hardinfo2", so that both `apt install hardinfo` and `apt install hardinfo2` will redirect to installing the `hardinfo` binary deb package.

Let me know if you have any comments or suggestions.

**Comment by lucascastro** [comment]
Em 20/05/2024 11:15, Boyuan Yang escreveu:
>
> So here is something new:
>
>   * I intend to take over the package maintenance of hardinfo (
>     https://tracker.debian.org/pkg/hardinfo ) and replace @tsimonq2
>     <https://github.com/tsimonq2> via Debian's internal process. This
>     is being tracked in https://bugs.debian.org/1071373 . Due to
>     timeout requirement set by Debian's policy, the maintenance
>     takeover will happen after 1 month if @tsimonq2
>     <https://github.com/tsimonq2> does not response.
>   * In the meanwhile, I don't mind spending some time to play around
>     hardinfo2's codebase and find any potential issues early.
>   * I am a Debian Developer, aka official Debian member.
>
What's your account/user at debian, it'll be required sponsorship.
>
>   * I could not find an ITP for hardinfo2 yet in Debian's database. cc
>     @lucascastro <https://github.com/lucascastro>
>
The ITP wasn't filed yet, because I'll make a last try to talk to tsimonq2.
>
>   * After that, I would like to cooperate with hardinfo2 upstream and
>     decide the best package name. Currently I think both "hardinfo" or
>     "hardinfo2" would be okay, depending on which name is the most
>     popular among Linux users. If you want, I can make package
>     "hardinfo" to provide "hardinfo2", so that both |apt install
>     hardinfo| and |apt install hardinfo2| will redirect to installing
>     the |hardinfo| binary deb package.
>
That's intend to be done if tsimonq2 not reply.

**Reply by hosiet** [reply]
My Debian account is https://nm.debian.org/person/byang/ . I don't know what do you mean by sponsorship, can you elaborate? If it is about getting new versions of hardinfo/hardinfo2 packge into Debian, I can do it by myself.

**Comment by lucascastro** [comment]
Em 20/05/2024 11:29, Boyuan Yang escreveu:
>
> My Debian account is https://nm.debian.org/person/byang/ . I don't 
> know what do you mean by sponsorship, can you elaborate? If it is 
> about getting new versions of hardinfo/hardinfo2 packge into Debian, I 
> can do it by myself.
>
The hardinfo2 package is already done, we are just trying to talk to 
hardinfo maintainer before upload.

**Reply by hosiet** [reply]
That would be great. When hardinfo2 gets into Debian, we can decide how to handle the old hardinfo package. Likely the outcome would be to remove the old package; it will be coordinated with hardinfo2 upstream of course.

**Reply by hwspeedy** [reply]
Thank you all so much for helping Debian branch get the hardinfo2 application into official repos.

**I have sent this via email to Simon Quiqley:**
> Hi Simon Quiqley,
> 
> I know you are really busy and doing a great job, but can you please hand the project over to debian developer - @hosiet (https://nm.debian.org/person/byang/), so that we do not have to wait a month for it to timeout if you do not answer.
> 
> https://bugs.debian.org/1071373
> 
> Thanx in advance and thanx for all you help with the hardinfo project in the past.
> 
> Tracking thread on github: https://github.com/hardinfo2/hardinfo2/discussions/16
> 


I will put both of you @lucascastro @hosiet on the Debian/Ubuntu maintainer task (One of you will be the official Debian/Ubuntu maintainer, you will have to work that out amongst yourselves)
 - The more eyes & hands - the better it works for users - which is our primary goal.
 
**Again - Thanx from me and all the Debian/Ubuntu branch users that requests the program!**

**Comment by hosiet** [comment]
Update on 2024-05-22:

* Simon replied in https://bugs.debian.org/1071373#10 , granting me the permission to make updates to Debian's [hardinfo](https://tracker.debian.org/pkg/hardinfo) package.
* I will make changes to update Debian's package maintainer information for `hardinfo` right now.

Now, we have an imminent issue that shall be determined ASAP: shall we continue to use the old package name `hardinfo`, or choose a new one `hardinfo2` as binary package name?

I am fine with either choice. It shall be determined by the current hardinfo2 upstream project members.

* If we would like to reuse `hardinfo`, I can update Debian's `hardinfo` to be hardinfo 2.x series with no time.
* If we would like to use the new `hardinfo2` name, I can still update Debian's `hardinfo` to be hardinfo 2.x series now. After that, I will perform extra steps internally in Debian to do package renaming. That will involve a thorough review by some other Debian people (i.e. the [Debian FTP Masters](https://ftp-master.debian.org/)), which will take some time before the renaming is completed.
* In either case, I can configure the package metadata so that users can install the package via both `apt install hardinfo` and `apt install hardinfo2`.

I can proceed with the above steps once we reach a conclusion to this issue.

**Reply by hwspeedy** [reply]
Project name is hardinfo2 (We had to add the "2", that has been there previously according to lpereira)

1. All other distro packages are made with hardinfo2 naming (But almost none of them had the old package).
2. The CPacks distributed in prereleases have package name hardinfo2 and breaks/replace hardinfo for DEB. (for RPM: obsoletes hardinfo)
3. If it can be installed as hardinfo/hardinfo2 and update hardinfo/hardinfo2 to hardinfo2, that would give a consistent naming as hardinfo2 (Matching github/webpage/branding) and be flexible for users.

I think you should decide for Debian/Ubuntu, you know the most and have experience in these cases.

**Comment by lucascastro** [comment]
Em 22/05/2024 15:20, Boyuan Yang escreveu:
>
> Update on 2024-05-22:
>
>   * Simon replied in https://bugs.debian.org/1071373#10 , granting me
>     the permission to make updates to Debian's hardinfo
>     <https://tracker.debian.org/pkg/hardinfo> package.
>   * I will make changes to update Debian's package maintainer
>     information for |hardinfo| right now.
>
> Now, we have an imminent issue that shall be determined ASAP: shall we 
> continue to use the old package name |hardinfo|, or choose a new one 
> |hardinfo2| as binary package name?
>
Good to know,

If I was going to maintain the package, it'd be prepared a transitional 
package, keep the source package name as hardinfo and update binary to 
hardinfo2 or hardinfo-ng.


> I am fine with either choice. It shall be determined by the current 
> hardinfo2 upstream project members.
>
>   * If we would like to reuse |hardinfo|, I can update Debian's
>     |hardinfo| to be hardinfo 2.x series with no time.
>   * If we would like to use the new |hardinfo2| name, I can still
>     update Debian's |hardinfo| to be hardinfo 2.x series now. After
>     that, I will perform extra steps internally in Debian to do
>     package renaming. That will involve a thorough review by some
>     other Debian people (i.e. the Debian FTP Masters
>     <https://ftp-master.debian.org/>), which will take some time
>     before the renaming is completed.
>   * In either case, I can configure the package metadata so that users
>     can install the package via both |apt install hardinfo| and |apt
>     install hardinfo2|.
>
> I can proceed with the above steps once we reach a conclusion to this 
> issue.
>
Just do it, no much thing to do yet.

Take a look at https://git.gnuabordo.com.br/hardinfo2.git

the package was already done.


Thanks,

Lucas Castro.

**Comment by hwspeedy** [comment]
Please try to avoid binary name "hardinfo-ng" - User support should be consistent across distros, if possible.

Otherwise sounds fine what Lucas says.

**Comment by lucascastro** [comment]
Em 22/05/2024 15:54, hwspeedy escreveu:
>
> Please try to avoid binary name "hardinfo-ng" - User support should be 
> consistent across distros, if possible.
>
> Otherwise sounds fine what Lucas says.
>
So, conclusion reached.

**Comment by hwspeedy** [comment]
@hosiet 
That was fast - thanx!
https://tracker.debian.org/pkg/hardinfo
https://ftp-master.debian.org/new/hardinfo_2.1.2-1.html

**recommends**
Can we upgrade suggests for 1:mesa-utils and 2:xdg-utils to recommends, please - This program has in the past had problems with not having the same support programs installed - we want the users to have the best experience.

1:Without mesa-utils - the GPU information will be missing and different from other distros.
2:Without xdg-utils the user cannot access the browser with links from program - homepage, bug reports, etc.

Please remove pciutils - it is only used on kernels not having sysfs. (>14years old distros)

See https://github.com/hardinfo2/hardinfo2/tree/master?tab=readme-ov-file#setting-up-addition-tools


**Debian bug/wish reports**
Remove all but 
487835 [Hardinfo] Phoronix & hardinfo?

Details:
#853749 [i|  |  ] [hardinfo] hardinfo gets segmentation fault -> FIXED ~7 crashes and alot more
#947159 [i|  |  ] [hardinfo] hardinfo segfaults when running benchmarks -> FIXED out of range crash
Outstanding bugs -- Normal bugs; Unclassified (7 bugs)
#523930 [n|  |  ] [hardinfo] [hardinfo] After some time it stops giving any kind of information -> FIXED OLD ERROR
#585685 [n|  |  ] [hardinfo] hardinfo: Sometimes hangs when using "Devices->Storage". Missed dependencies. -> FIXED udisks2
#638951 [n|  |  ] [hardinfo] hardinfo: does not show usb devices -> FIXED sysfs
#658985 [n|  |  ] [hardinfo] hardinfo: benchmark graphic bar lengths confusing and undocumented. -> FIXED
#721122 [n|  |  ] [hardinfo] hardinfo: unable to generate or/and save reports -> FIXED OLD ERROR
#967518 [n|  |♔] [src:hardinfo] hardinfo: depends on deprecated GTK 2 -> FIXED GTK3
#1014266 [n|  |  ] [src:hardinfo] hardinfo: Consider packaging new upstream snapshot? - FIXED when hardinfo2 released.
Outstanding bugs -- Minor bugs; Unclassified (4 bugs)
#492989 [m|  |  ] [hardinfo] hardinfo: doesn't use gnome preferred browser -> FIXED xdg_open
#524844 [m|  |  ] [hardinfo] hardinfo: benchmark results unreadable -> FIXED GTK3
#572799 [m|  |  ] [hardinfo] hardinfo: dipslays SATA drives as SCSI -> FIXED sysfs
#886280 [m|  |  ] [hardinfo] hardinfo: Change icon in .desktop file -> FIXED .desktop file
Outstanding bugs -- Wishlist items; Patch Available (1 bug)
#1051775 [w|+|  ] [hardinfo] hardinfo: Please add support for hardinfo -> FIXED loongarch
Outstanding bugs -- Wishlist items; Unclassified (1 bug)
#486471 [w|  |  ] [hardinfo] [hardinfo] library for icons which are in hardinfo and tango? -> CLOSE WON'T FIX
Forwarded bugs -- Minor bugs (1 bug)
#483136 [m|  |↝] [hardinfo] [hardinfo] shows only one temperature-sensor -> FIXED
Forwarded bugs -- Wishlist items (6 bugs)
#446613 [w|  |↝] [hardinfo] gpu benchmark -> DONE
#455568 [w|  |↝] [hardinfo] hardinfo: 'Storage>IDE Disks>CD-RW 52x24>Speeds' gives puzzling DVD speed. -> FIXED OLD ERROR
#455569 [w|  |↝] [hardinfo] hardinfo: Please add sorting to 'Computer>Kernel Modules', et al. -> FIXED
#458507 [w|  |↝] [hardinfo] improve benchmark-system -> FIXED new methods
#458838 [w|  |↝] [hardinfo] benchmarks: marking average -> FIXED on webpage - lot more to come.
#487835 [w|  |↝] [hardinfo] [hardinfo] phoronix & hardinfo? -> **ON WISH LIST - LEAVE OPEN**

**Comment by hwspeedy** [comment]
@hosiet

**Package Recommends**
Please also add fwupd to recommands.
This allows the user to see firmware versions, which is more and more important due to security issues.

Updated README.md with package installed packages:
See https://github.com/hardinfo2/hardinfo2/tree/master?tab=readme-ov-file#setting-up-addition-tools

**Comment by hwspeedy** [comment]
@hosiet 

Please change control architecture to linux only - we do not support hurd.

**Architecture: linux-any**

**Comment by hosiet** [comment]
A new Debian upload containing the 2025-05-25 snapshot is now in Debian Sid/Unstable. Please check https://tracker.debian.org/pkg/hardinfo for more information.

I am wondering if it is okay for Debian's package to provide a compatibility symlink (`/usr/bin/hardinfo` -> `/usr/bin/hardinfo2`) in the `hardinfo` package. I believe such symlink would be harmless, and can help old users to migrate to the new codebase more smoothly.

**Comment by hwspeedy** [comment]
@hosiet 

**Very fast work, thanx.**

We have some issues yet - we need a new -2 build before next release to look at buildlogs.

1. 4 patches for building all linux architectures **(DONE - Sparc64 does not work - cannot see any code errors - @hosiet )**
https://github.com/hardinfo2/hardinfo2/issues/40

2. Recommends packages **(DONE - Missing remove pciutils @hosiet )**
control file - you have updated the 2 suggests to recommends, please add fwupd to recommends and remove pciutils

3. Build for all linux architectures (not hurd). **(DONE)**
Change architecture to linux-any in control file

4. Error in cli report **(DONE)**
I will look into the found issue https://github.com/hardinfo2/hardinfo2/issues/4, that needs fixing before next release also.
patch: https://github.com/hardinfo2/hardinfo2/commit/30d9c3f5c4498bf0b8338ea9f97490425237ed88

5. sym-link should either be in github repo or not at all. How many have actually the old program installed? - seams it was pretty dead, so don't see the need for symlink, as most probably start from GUI. How do you feel?

6. Debian bugtracker needs to be cleaned according to this discussion. **(DONE)**


PS: It's very important to get this first release right as it will be cloned to alot of distros....

SALSA: https://salsa.debian.org/debian/hardinfo/-/blob/master/debian/control?ref_type=heads

**Reply by hosiet** [reply]
That's okay, I don't mind frequent releases. Recently I will be packaging several snapshots in Debian to make tests sort out all issues.

**Reply by hosiet** [reply]
If you want to take a look at existing installations, please check the history data as shown on https://qa.debian.org/popcon.php?package=hardinfo .

**Reply by hwspeedy** [reply]
Yes, It was so populare and such a nice program back then - but due to a LIFE - it's now place 71913. 
I hope it will get it's glory back and live for a long time...

We will only distribute releases.

PS: Updated above todo list with status from your salsa.

**Reply by hosiet** [reply]
For architecture support in Debian -- I don't care sparc64 since it is not an official supported arch in Debian. If it fails to build, I would rather let it stands there until anyone interested would like to handle it.

For pciutils, it has been demoted to Suggests: field as of https://salsa.debian.org/debian/hardinfo/-/blob/2b7d6c4b6181599bcaefd155c6c75f2b5d0bdba1/debian/control#L50 . Packages listed in Suggests: field is basically useless so keeping it there should be fine.

The bug tracker has already been cleaned up. If you want, check individual reports.

**Comment by hwspeedy** [comment]
Okay, sounds good - Sparc64 accepted as non important.

I have just submitted patch for #41.

Are we pretty confident? - then I can do a release in 2 hours - I test build for ~20 targets before release.

**Reply by hosiet** [reply]
I believe current code should be okay.

**Comment by hwspeedy** [comment]
@hosiet - release 2.1.5 ready

**Comment by hwspeedy** [comment]
@hosiet - Please check that you have CMAKE_BUILD_TYPE=Release, thanx.

**Reply by hosiet** [reply]
While I understand that you may want to use `CMAKE_BUILD_TYPE` to control certain things, Debian in general is against a preset `CMAKE_BUILD_TYPE` value, especially not the `Release`.

I found a similar case at https://bugs.debian.org/730636 . The whole story there should be similar to what we are facing in hardinfo2 now.

**Reply by hwspeedy** [reply]
@hosiet - Okay, not sure what the problem is for Debian - Fedora/Arch etc. uses Release, so it must be something in internal build tools...
 
Distro builds: CMAKE_BUILD_TYPE=Release
User/GitHub builds: CMAKE_BUILD_TYPE=None 
Developers use CMAKE_BUILD_TYPE=Debug for adding additional tools like asan and Debug output.

I believe cmake changes to optimize for Release vs None.

On the server, debug benchmarks are discarded, Release benchmarks have higher trust, as these are build by PROs.

**Comment by lucascastro** [comment]
Please, add Provides: hardinfo2 at debian packge and close #1071528 bug.

Em 25/05/2024 18:00, hwspeedy escreveu:

**Reply by hosiet** [reply]
Debian Bug #1071528 is now closed.

I don't think adding any "Provides: hardinfo2" is necessary. End users do not care what source package name is. Users only see the binary package names. All they need is to do `apt install hardinfo2`, which is already available with the current setup.
